### PR TITLE
feat: module for ldap authentication

### PR DESCRIPTION
**Pull request by ireznice**
first attempt how it may looks like.

### CODE REVIEW

File: web/middleware/auth_ldap.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    pass
+
+async def parse_auth(request):
+    auth_string = request.headers['authorization']
+    if auth_string.find("Basic ") == 0:
+        hash_string=auth_string[6:]
+        base64_bytes = hash_string.encode('ascii')
+        message_bytes = base64.b64decode(base64_bytes)
+        message = message_bytes.decode('ascii')
+        mess_arr = message.split(':')
```
Comment (petrSchreiber) [inline web/middleware/auth_ldap.py:34]: Hi Ivo,

please restrict the split here to happen just once, by using message.split(':', 1)

File: webserver.py [hunk 2]
```diff
@@ -15,11 +15,12 @@
...
-                                                    sanic,
-                                                    hex(id(loop)),
-                                                    threading.current_thread().name
-        ))
+        pass
+#        logger.debug('before_start {} {} {}'.format(
+#                                                    sanic,
+#                                                    hex(id(loop)),
+#                                                    threading.current_thread().name
+#        ))
```
Comment (petrSchreiber) [inline webserver.py:23]: I would prefer to not keep unused code any longer.

File: web/middleware/auth_ldap.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        return None
+
+async def check_group(conn, group_dn, user_dn):
+    result = conn.search_s(group_dn, ldap.SCOPE_BASE, filterstr="(&(objectClass=group)(member={}))".format(user_dn), attrlist=["name"])
+    #print(result)
+    try:
+        if result[0][0] == group_dn:
+            return True
+        print(result)
+        return False
```
Comment (petrSchreiber) [inline web/middleware/auth_ldap.py:55]: Marked section can be simplified as:
```
logger.debug(f'check_group -> {result}')
return result[0][0] == group_dn
```

File: webserver.py (hunk 2 above)
Comment (ireznice) [reply webserver.py]: removed

**Review by petrSchreiber — APPROVED** [review]
Thank you for the adjustments.

File: web/middleware/auth_ldap.py (hunk 1 above)
Comment (petrSchreiber) [reply web/middleware/auth_ldap.py]: Addressed

**Pull request merged**